### PR TITLE
ref-range-fasta command

### PR DESCRIPTION
**Pull request by tcasstevens**
## Description

This creates a fasta for each ReferenceRange specified. One issue remaining is if multiple samples have the same sequence (i.e. checksum / ALT Header id), all but one ALT header is lost in the merged HaplotypeGraph.  Therefore, some descriptions in the output Fastas can show the wrong sample / positions in the descriptions

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [X] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: src/main/kotlin/net/maizegenetics/phgv2/cli/RefRangeFasta.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import net.maizegenetics.phgv2.api.HaplotypeGraph
+import net.maizegenetics.phgv2.api.ReferenceRange
+import net.maizegenetics.phgv2.utils.Position
+import net.maizegenetics.phgv2.utils.loadRanges
+import net.maizegenetics.phgv2.utils.seqFromAGC
+import org.apache.logging.log4j.LogManager
+import java.io.File
+
+/**
+ * Write one Fasta file reference range specified by the bedfile.
```
Comment (lynnjo) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/RefRangeFasta.kt:17]: Add the word "per" in this sentence to make it clearer (I had trouble understanding at first).  "Write one Fasta file per reference rnage specified by the bedfile"

File: src/main/kotlin/net/maizegenetics/phgv2/utils/SeqUtils.kt [hunk 2]
```diff
@@ -385,3 +387,88 @@ fun queryAgc(commands:Array<String>):Map<Pair<String,String>,NucSeq> {
     }
     return genomeChromNucSeq
 }
+
```
Comment (lynnjo) [inline src/main/kotlin/net/maizegenetics/phgv2/utils/SeqUtils.kt:390]: Add function description comment before the function.

**Review by smm477 — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/RefRangeFasta.kt (hunk 1 above)
Comment (tcasstevens) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/RefRangeFasta.kt:17]: Ok I will add that in the refactored version

File: src/main/kotlin/net/maizegenetics/phgv2/utils/SeqUtils.kt (hunk 2 above)
Comment (tcasstevens) [reply src/main/kotlin/net/maizegenetics/phgv2/utils/SeqUtils.kt:390]: Ok, I will do that

**Comment by lynnjo** [comment]
@tcasstevens I know you made changes to CreateFastaFromHvcf.kt as I can see them when I load your branch into Intellij.  But those do not show up in this pull request - I don't know why.

My comment on that file:  update the help message for the fastaType parameter to include pangenomeHaplotype and rangeFasta

**Review by lynnjo — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateFastaFromHvcf.kt [hunk 3]
```diff
@@ -75,7 +80,7 @@ class CreateFastaFromHvcf : CliktCommand(help = "Create a FASTA file from a h.vc
         rangeFasta
     }
 
-    val fastaType by option("--fasta-type", help = "Type of fasta exported.  Can be either composite or haplotype")
+    val fastaType by option("--fasta-type", help = "Type of fasta exported.")
```
Comment (lynnjo) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/CreateFastaFromHvcf.kt:83]: Should the help message be more specific to mention what types?  If given a list, the user could investigate further if they didn't understand what each type represented.

**Review by zrm22 — APPROVED** [review]

**Pull request merged**